generic/BRGenericHandlers.h

```diff
@@ -33,6 +33,10 @@ extern "C" {
         BRGenericAccount base;
     } *BRGenericAccountWithType;;
 
+    typedef struct {
+        BRGenericHandlers handlers;
+    } BRGenericNetworkRecord;
+
```

Comment:
This is okay for now but is something that needs consideration.

Notice that `BRGenericAddress`, `BRGenericWallet`, `BRGenericTransfer`, .. do not have corresponding structures  at the 'generic' level.  Instead address, wallet, transfer, ... are simply `void*` because they come from the 'ripple', 'stellar', ... level.  

What is the reason that `BRGenericAccountWithType` and `BRGenericNetworkRecord` need to exist?  It is because in these cases the `handlers` need to be accessed.  Normally the access to `handlers` is through BRGenericWalletManager.  But 'account' and 'network' can't know about a 'wallet manager' because 'account' and 'network' pre-exist 'wallet manager' and, in fact, are need to create a 'wallet manager'.

There is also a future issue of thread safety (lock/unlock).  The 'wallet manager' holds a lock that protects its contents including access to the 'wallet managers' wallet(s), transfers, etc.  Do 'account' and 'network' need a lock as well?

I'm actively wondering:

1. Perhaps these `typedef void*` declarations need to be replaced with a more uniform design like:
```
typedef struct ... {
  BRGenericHandlers handler;
  void *underlyingRippleStellar...Reference;
} BRGeneric<Thing>;
```
which, by holding `handlers`, allows the context for the underlying reference to be understood.
2. Or the CWM level, for things like 'account' and 'network' need to hold not only the 'void*' thing but also a 'handers' thing so that the context isn't lost.
3. Accept that 'account', 'network' and 'wallet manager' won't really be 'typedef void*', they will have some structure at the 'GEN' level, unlike wallet, transfer, etc.

@carlcherry, @michael-brd Am I making sense?  Thoughts?

Reply:
And more, the case of `Network` is an interesting one.  We've come to the conclusion, I think, that networks are built in, in some form.  Core software comes configured for {BTC Mainnet, BTC Testnet, BCH Mainnet, BCH Testnet, ETH Mainnet, ETH Ropsten, XRP Mainnet} - because we have source and configurations (e.g. ChainParams) built in.  Perhaps BRCryptoNetwork instances need to be 'statically defined' for the builtin networks/blockchains.

And 'Account' is also interesting.  In order to serialize an 'account' (tso the IOS/Android App can use it on restart) we *must* know about the built-in blockchains - because we *must* generate a public address from the BIP39 seed, while we have the seed and before use/serialization.

Reply:
I wonder if the statically defined bits should be split off into a BRCryptoBlockchain type (or what have you, which a BRCryptoNetwork would have). The BRCryptoBlockchain "class" would be immutable ("source and configurations"), while the network would have the mutable properties. The built-ins would then be of the BRCryptoBlockchain type, rather than BRCryptoNetwork.

That would provide a clean separation of responsibilities, rather than having built-in networks that we are not meant to mutate (i.e. cryptoNetworkSetHeight/cryptoNetworkSetNetworkFees).

Things like Account could then make due with the built-in blockchains.

Reply:
Copied above to CORE-541.